src/shogun/machine/gp/KLDualInferenceMethod.h

```diff
@@ -39,9 +39,52 @@
 #include <shogun/lib/config.h>
 #include <shogun/machine/gp/KLInference.h>
 #include <shogun/machine/gp/DualVariationalGaussianLikelihood.h>
+#include <shogun/optimization/lbfgs/LBFGSMinimizer.h>
 
 namespace shogun
 {
+
+class KLDualInferenceMethodMinimizer: public LBFGSMinimizer
+{
+public:
+	KLDualInferenceMethodMinimizer(): LBFGSMinimizer() { init(); }
+  
+	KLDualInferenceMethodMinimizer(FirstOrderCostFunction *fun): LBFGSMinimizer(fun) { init(); }
+  
+	virtual ~KLDualInferenceMethodMinimizer() {}
+  
+	virtual float64_t minimize();
+  
+protected:
+	/** Init before minimization */
+	virtual void init_minimization()
+	{
+		REQUIRE((m_linesearch == ELBFGSLineSearch::BACKTRACKING_ARMIJO) ||
+			(m_linesearch == ELBFGSLineSearch::BACKTRACKING_WOLFE) ||
+			(m_linesearch == ELBFGSLineSearch::BACKTRACKING_STRONG_WOLFE),
+			"The provided line search method is not supported. Please use backtracking line search methods\n");
+		LBFGSMinimizer::init_minimization();
+	}
```

Comment:
Is there a strong reason to keep implementation of these methods in the header? Also for the ones below.